feat: show dropdown with selected items when readonly

web-padawan opened the pull request:
## Description

As discussed internally, the dropdown should be also available when readonly, so user can see all the selected items.
In this case, interacting with the items isn't possible: you can't un-select them with keyboard or mouse click.

![readonly-dropdown](https://user-images.githubusercontent.com/10589913/163784624-6177a1f5-3daf-4411-804c-9616d5164e3b.gif)

## Type of change

- Feature

jouni commented:
If I can wish, I would also hide the clear button in the chips (`x`) in read-only mode.

web-padawan commented:
> I would also hide the clear button in the chips (x) in read-only mode.

Thanks, done. Also updated themes to hide it when disabled for consistency.

vursen commented:
Just happened to notice a strange thing in the gif. What is this?

<img width="250" alt="image" src="https://user-images.githubusercontent.com/5039436/163963411-4ea67afa-e92f-4978-880e-5eedfec1d9d9.png">

jouni commented:
> What is this thing?

It’s a visual indication that there are multiple "collapsed/stacked" chips, not just one.

web-padawan commented:
> Just happened to notice a suspicious artifact in the gif. What is this thing?

Actually, it's kind of a feature, related to another PR in review - please see #3632.

Here is the original design for this extra border: https://github.com/vaadin/platform/issues/2690#issuecomment-1028951117

vursen commented:
> It’s a visual indication that there are multiple "collapsed/stacked" chips, not just one.

> Actually, it's kind of a feature, related to another PR in review - please see https://github.com/vaadin/web-components/pull/3632.

Ah, right. :)

DiegoCardoso commented:
I am thinking that the toggle button color could be in a different shade. Maybe the same as the color when it's not on ready only state. My feeling is that with the current color, it doesn't clearly indicate it's clickable (and might even fail some a11y contrast test, but I didn't test it yet).

web-padawan commented:
> the toggle button color could be in a different shade. Maybe the same as the color when it's not on ready only state

Good catch. Updated Lumo and Material themes accordingly.

DiegoCardoso commented:
For a multi select with no items selected, the overlay opens with all the options.

https://user-images.githubusercontent.com/262432/164031633-1484f951-5030-4103-af6f-bb021cb104d8.mp4

web-padawan commented:
> For a multi select with no items selected, the overlay opens with all the options.

@DiegoCardoso Thanks, fixed. Also updated toggle button styles to use `[has-value]` accordingly.

DiegoCardoso commented:
nit: maybe we can change the cursor to default here

https://user-images.githubusercontent.com/262432/164391564-61836cb3-4d4e-4e2e-9888-56215602549b.mp4

.

web-padawan commented:
> maybe we can change the cursor to default here

@DiegoCardoso Thanks, fixed.

knoobie commented:
@web-padawan looks like this feature doesn't work (can be reproduced in the docs and was reported here https://github.com/vaadin/platform/issues/3257)

web-padawan commented:
Thanks. It's also reported here: https://github.com/vaadin/flow-components/issues/3684. We will look into it.
